go/extractor/util/registryproxy.go

```diff
@@ -0,0 +1,140 @@
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+const PROXY_HOST = "CODEQL_PROXY_HOST"
+const PROXY_PORT = "CODEQL_PROXY_PORT"
+const PROXY_CA_CERTIFICATE = "CODEQL_PROXY_CA_CERTIFICATE"
+const PROXY_URLS = "CODEQL_PROXY_URLS"
+const GOPROXY_SERVER = "goproxy_server"
+
+type RegistryConfig struct {
+	Type string `json:"type"`
+	URL  string `json:"url"`
+}
+
+// The address of the proxy including protocol and port (e.g. http://localhost:1234)
+var proxy_address string
+
+// The path to the temporary file that stores the proxy certificate, if any.
+var proxy_cert_file string
+
+// An array of registry configurations that are relevant to Go.
+// This excludes other registry configurations that may be available, but are not relevant to Go.
+var proxy_configs []RegistryConfig
```

Comment:
These no longer need to be global variables. I think it would be clearer if they were just local variables.

Reply:
I somewhat intentionally left these as global variables so that we can access their values elsewhere if needed going forward.

I don't feel strongly about this though, so if you would prefer it to be locals for now while they don't have to be globals, then I can make that change.